### ISSUE TITLE
Media Player Accessibility / assessment

### ISSUE THREAD

**Issue by btopro** [elmsln/elmsln#96]
We've been fighting against MediaElement.js it pains me to say. ELMSMedia currently uses it to display the video and it's just got a lot of integration problems.  No deal breakers, but things we are better served putting efforts elsewhere.

OSU did an audit of the accessibility of video players: https://docs.google.com/spreadsheets/d/1QJVcXx5hTWYBcJbHJD3DrL3hSFVbfy1VQFyADMtrDFY/edit#gid=0

Very comprehensive, and as you can see there are no perfect options. The best option is of course the worst looking one, Able Player: http://ableplayer.github.io/ableplayer/tests/

There's a Drupal sandbox I'm going to test shortly that supposedly integrates it: https://www.drupal.org/sandbox/afemath/2475577

**Comment by kfurst** [comment]
"The best option is of course the worst looking one, Able Player"

Agreed there. Yet is there a triage that needs to be applied here? While I would agree that the highest level of accessibility would be the goal, I would also argue that when it comes to go-to content providers, there are some heavy hitters on that audit spreadsheet and then there are some I've never heard of/can't think of a provider/source that uses that particular player/approach. To me, there is a weight to the design component. While IT guys might argue, "Who cares about what the bar looks like as long as the thing works?" I would tell you the answer is: Most teachers and kids I've worked with. Seriously. 

I'm always amazed by how inviting design and overly simplistic function end up being lynchpins in so many of my current LMS related meetings (or arguments) with faculty, students, in regards to Moodle. (BTW... Moodle sucks at both of those). I would add also that the K12 staff I work with grab content from an extremely short list of providers... 

I would say design/form of the player weighs heavily in my mind when wrestling with this specific issue. So as you say @btopro JustWorks™ but add ©ItsBeautiful

**Comment by btopro** [comment]
uhh... @michael-collins can make things look great. Able Player is open source. Most accessible player + Open source + Great designer = We'll have the best video player.

**Comment by michael-collins** [comment]
@kfurst I suspect you'll enjoy this paper: http://www.westga.edu/~distance/ojdla/winter134/david_glore134.html

**Comment by kfurst** [comment]
@btopro Oh...Well, then is there are reason not to go with AblePlayer that I'm missing? It seems to have the basics required for what's needed. It seems like you're totally right in going down that road then. 

Though... I would also say that it's somewhat of a disappoint to see the priority 4 status of everything that would reduce the unidirectional nature of content. I know there are a few 3rd party products that have hit the market recently that have tried to address this and of the three I've tried... well... not impressed. My point = In my dreamworld, I would love to have a cool player that provided some more dynamic interactivity with content beyond glorified commenting.

To that end, Able seems to have options relating to playback speed, annotations, timers, etc. Those ARE very important features for those I work with that use a lot of video in house (same for Univ?) 

@michael-collins   Had that one in your back pocket ready to go?:-)

 paper + me = agreed (and a lot of those citations date to 02 and 04 when everything was ugly).

**Comment by michael-collins** [comment]
@kfurst As of the other day, it is permanently in my back pocket. ;)

**Comment by btopro** [comment]
@kfurst So this player goes in the media system for the boring stuff (aka lectures, the things students don't pay attention to in class so online left to their own devices they'll totally dig!). What you describe is more of a learning object to me then a video outright (I think, please correct me if I'm wrong). I think @rickhumphries would say that this capability is being brought it via H5P. We'll be focusing on that / we need someone to focus on, the distribution I refer to as "interact".

Interact would be for interactive assets and things that potentially ingest responses / allow for branching paths. We've got some really rough prototypes in there (we being my group and @rickhumphries ), him with H5P as a interactive, mini assessment / quiz engine (which does have support for video / stop and answer questions, etc). We've got some interactive timesline via timelinejs and a few SmartBuilder applications integrated.

If your thinking something outside this idea, please post another issue related to this idea and mark it as pedagogy cause we need all the voices in that category that we can :). I've also gotten my IDs cool with contributing to conversation in this space so that'd be a great conversation for them to have too.

**Comment by kfurst** [comment]
@btopro 
1.Ahh... that makes sense (ignorance on my part, I've got to locate some of the threads that sum this up. Somewhat struggling on getting up to speed on important distinctions like this via existing issue threads) and dead on with H5P. 
2. Will do (next week) re: Interact, Pedagogy, etc.
3. The Interact module will be one of the biggest "things" for me since A. It takes the right kind of person with serious chops to lecture effectively and B. It's rare to find A. in K-12. Thus K-12's focus (albeit not so great so far) on progressive pedagogy and inductive, inquiry-based ed. Yet, our LMS options lean towards traditionalist approaches which I'm very wary of falling back on. 

So...umm... great. Done now. Promise.

**Comment by btopro** [comment]
H5P now comes OOTB with the media system so at least one of these things is being addressed so far. now for the rest :)

**Comment by btopro** [comment]
This sandbox is absolutely worth aligning with, building upon and figuring out how we can fully integrate it with elmsmedia system -- https://www.drupal.org/sandbox/npacker/2508441 . I had a prototype of flex-video working after just a few minutes / DOM hacking. We'd need to add some support for additional buttons / flexibility added to the UI in the custom settings but it's most of the way along and done "The Right Way".

**Comment by btopro** [comment]
http://ableplayer.github.io/ableplayer/tests/test7.html @michael-collins

**Comment by btopro** [comment]
@kat-wehr @curiosity26 If we can't use the same system for video but are both using Ableplayer we should talk styling / ideas around flexibility in settings / etc as it's designed for people

**Comment by btopro** [comment]
@heyMP next time we get together this branch is the priority. I need to step through some things with you since you know a lot more about the display tab / modes then I do so I want to confirm the settings that I have for them. Ableplayer is working though could stand to have some additional flexibility as far as what options are included but wow does the 3.x version they just released look better then past versions. Highly recommend checking it out in a VM to see.

**Comment by btopro** [comment]
it's pretty awesome looking. I still need to do more testing before we can merge this branch in and do an upgrade hook but here's a screenshot.

![2016-07-31_21-38-41](https://cloud.githubusercontent.com/assets/329735/17381752/893ff00a-599b-11e6-8e4f-7a5652a536d9.png)

**Comment by btopro** [comment]
this isn't as flexible as I'd like but basically it just needs testing with a real setup to close / an upgrade type

**Comment by btopro** [comment]
This needs production testing and additional VM spin ups to verify working out of the box. It'll need to see future API flexibility but it's ready for prime time! :)
